cli_tools/gce_ovf_import/ovf_importer/ovf_param_validator.go

```diff
@@ -0,0 +1,255 @@
+//  Copyright 2019 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package ovfimporter
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"google.golang.org/api/iterator"
+
+	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/common/domain"
+	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/common/utils/compute"
+	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/common/utils/logging"
+	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/common/utils/param"
+	pathutils "github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/common/utils/path"
+	storageutils "github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/common/utils/storage"
+	"github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/common/utils/validation"
+	ovfdomain "github.com/GoogleCloudPlatform/compute-image-tools/cli_tools/gce_ovf_import/domain"
+	"github.com/GoogleCloudPlatform/compute-image-tools/daisy"
+)
+
+const (
+	// InstanceNameFlagKey is key for instance name CLI flag
+	InstanceNameFlagKey = "instance-names"
+
+	// MachineImageNameFlagKey is key for machine image name CLI flag
+	MachineImageNameFlagKey = "machine-image-name"
+
+	// MachineImageStorageLocationFlagKey is key for machine image storage location CLI flag
+	MachineImageStorageLocationFlagKey = "machine-image-storage-location"
+
+	// ClientIDFlagKey is key for client ID CLI flag
+	ClientIDFlagKey = "client-id"
+
+	// OvfGcsPathFlagKey is key for OVF/OVA GCS path CLI flag
+	OvfGcsPathFlagKey = "ovf-gcs-path"
+
+	// HostnameFlagKey is key for hostname CLI flag
+	HostnameFlagKey = "hostname"
+)
+
+// ParamValidatorAndPopulator validates parameters and infers missing values.
+type ParamValidatorAndPopulator struct {
+	metadataClient        domain.MetadataGCEInterface
+	zoneValidator         domain.ZoneValidatorInterface
+	bucketIteratorCreator domain.BucketIteratorCreatorInterface
+	storageClient         domain.StorageClientInterface
+	logger                logging.Logger
+}
+
+// ValidateAndPopulate validates OVFImportParams, and populates values that are missing.
+// It returns an error if params are invalid.
+func (p *ParamValidatorAndPopulator) ValidateAndPopulate(params *ovfdomain.OVFImportParams) (err error) {
```

Comment:
Most of the `determine*` functions are doing validation and value building. `Determine` doesn't seem like the right verb for these two operations.

Reply:
Good point. Updated the functions with more descriptive name (such as `calculateDeadlineFromTimeout`)